File: Assignments/Acme-Electronics-Shop/CustomerApp/app/src/main/java/pt/up/feup/cpm/customerapp/models/Product.kt
package pt.up.feup.cpm.customerapp.models

class Product {
    private var productID: String? = null
    private var name: String? = null
    private var price = 0.0
    private var characteristics: String? = null
    private var make: String? = null
    private var model: String? = null

    constructor(productID: String?, name: String?, price: Double, characteristics: String?, make: String?, model: String?) {
        this.productID = productID
        this.name = name
        this.price = price
        this.characteristics = characteristics
        this.make = make
        this.model = model
    }

    fun getProductID() : String? {
        return this.productID
    }

    fun getName() : String? {
        return this.name
    }

    fun getPrice() : Double? {
        return this.price
    }

    fun getCharacteristics() : String? {
        return this.characteristics
    }

    fun getMake() : String? {
        return this.make
    }

    fun getModel() : String? {
        return this.model
    }
}

File: Assignments/Acme-Electronics-Shop/Remote Service/models/Product.js
import mongoose from "mongoose";

export const productSchema = new mongoose.Schema({
    productID: { type: String, required: true },
    name: { type: String, required: true },
    price: { type: Number, required: false },
    make: { type: String, required: false },
    model: { type: String, required: false },
    characteristics: { type: String, required: false }
});

let Product = mongoose.model("Product", productSchema);

export default Product;